From cec59f09f511cf77171074f3bb0d0ce043835ae3 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 21 Nov 2014 13:35:11 +0530
Subject: [PATCH 1/1] EM/Maxim: Treat battery as invalid if charge profile type
 is CHRG_PROF_NONE.

Curretly, the battery is treated as invalid only if no charge profile is
inserted to charger framework. But the battery should be treated as invalid
even if charger profile is present and its type is CHRG_PROF_NONE. This patch
will fix this issue.

Jira-Id: https://jira01.devtools.intel.com/browse/AKP-669
Change-Id: I5b39b6fff7d57f6783456de1a27c984fb616cdc2
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/power/max17042_battery.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/power/max17042_battery.c b/drivers/power/max17042_battery.c
index 7403ff9..e815211 100644
--- a/drivers/power/max17042_battery.c
+++ b/drivers/power/max17042_battery.c
@@ -910,8 +910,12 @@ max17042_get_pdata(struct device *dev)
 		return NULL;
 
 #ifdef CONFIG_POWER_SUPPLY_CHARGER
+	memset(&batt_prof, 0 , sizeof(batt_prof));
 	ret = get_batt_prop(&batt_prof);
-	if (ret < 0) {
+	/* Treat the battery as invalid if charge profile not found
+	 * or type is CHRG_PROF_NONE.
+	 */
+	if (ret < 0 || batt_prof.chrg_prof_type == CHRG_PROF_NONE) {
 		pdata->enable_current_sense = false;
 		snprintf(pdata->battid, (BATTID_LEN+1),
 			"%s", "UNKNOWNB");
-- 
1.9.1

